scripts/alert_emails_report/generate-report.ts

```diff
@@ -0,0 +1,134 @@
+import path from 'path';
+import _ from 'lodash';
+import moment from 'moment';
+import CampaignMonitor, {
+  CampaignMonitorStats,
+} from '../alert_emails/campaign-monitor';
+import {
+  getFirestore,
+  fetchAllAlertSubscriptions,
+} from '../alert_emails/firestore';
+import GoogleSheets, { Cell } from '../common/google-sheets';
+import {
+  getLocationNameForFips,
+  findCountyByFips,
+  isStateFips,
+} from '../../src/common/locations';
+import { ALERT_EMAIL_GROUP_PREFIX } from '../alert_emails/utils';
+
+const SPREADSHEET_ID = '1cs3Wyh8Gda0H18_-x5RYp7AJ3FwB-I1ewAQTBWEk1YY';
+const REPORT_URL = `https://docs.google.com/spreadsheets/d/${SPREADSHEET_ID}`;
+
+const keyFile = path.join(
+  __dirname,
+  '../alert_emails/google-service-account.json',
+);
+
+interface FipsCount {
+  fips: string;
+  count: number;
+}
+
+async function main() {
+  try {
+    await updateSubscriptionsByLocation();
+  } catch (err) {
+    console.error('Error updating subscription stats', err);
+  }
+
+  try {
+    await updateEngagementStats();
+  } catch (err) {
+    console.error('Error updating engagement stats', err);
+  }
```

Comment:
This error handling feels a bit non-ideal.
1. main() is declared `async` which means it returns a `Promise`, but the calling code doesn't handle the Promise, which seems a bit misleading to me (not a huge deal though as long as we make sure that main()'s error handling is bulletproof).
2. I'm not really sure why you have 2 separate try/catch blocks.  Shouldn't the stack trace be enough to tell what failed?
3. Your catch block is _logging_ the error, but the script will still exit with error code 0 as if it was successful, which means if we run this script in a github workflow or whatever, the workflow will "succeed" even though there was an error, which means we probably won't notice it.

What I've done in other scripts is:

`main().catch(e => { console.error(e); process.exit(1); });`

or similar...  You don't have to do the same, but it would address all my feedback. 😁